File: backend/spring-nfc/src/main/java/com/example/springnfc/entity/Installation.java
package com.example.springnfc.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Collection;
import java.util.Date;


@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@ToString
public class Installation implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    @Lob
    private String description;
    private String adress;
    private String city;
    private String zipcode;
    private String image;

    private Date createdAt = new Date();
    private Date updatedAt;

    @ManyToOne
    private Client client;

    @ManyToOne
    private Utilisateur utilisateur;

    @OneToMany(mappedBy="installation")
    private Collection<InterventionProvider> interventionList;

    @OneToMany(mappedBy="installation")
    private Collection<Intervention> interventions;
}

File: frontend/flutter-nfc/lib/models/installation.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
class Installation {
  int? id;
  dynamic title;
  dynamic adress;
  dynamic description;
  dynamic city;
  dynamic zipcode;
  dynamic image;

  Installation({
    this.id,
    this.title,
    this.adress,
    this.description,
    this.city,
    this.zipcode,
    this.image,
  });

  Installation.fromJson(Map<String, dynamic> json) {
    id = json["id"] ?? '';
    title = json['title'] ?? '';
    adress = json['adress'];
    description = json['description'] ?? '';
    city = json['city'] ?? '';
    zipcode = json['zipcode'] ?? '';
    image = json['image'] ?? '';
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['title'] = title.toString();
    data['adress'] = adress.toString();
    data['description'] = description.toString();
    data['city'] = city.toString();
    data['zipcode'] = zipcode.toString();
    data['image'] = image.toString();

    return data;
  }
}
